experiment/secret-sync-controller/client/client.go

```diff
@@ -0,0 +1,48 @@
+package client
+
+import (
+	"context"
+	"flag"
+	"os"
+	"path/filepath"
+
+	secretmanager "cloud.google.com/go/secretmanager/apiv1"
+	"google.golang.org/api/option"
+
+	"k8s.io/client-go/kubernetes"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func NewK8sClientset() *kubernetes.Clientset {
+	var kubeconfig *string
+	if home := os.Getenv("HOME"); home != "" {
+		kubeconfig = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+	} else {
+		kubeconfig = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
+	}
+	flag.Parse()
```

Comment:
If you're trying to set this up as a command line flag, you should do so in the `options` struct and follow the pattern you used for `config-path`, then turn this into `NewK8sClientset(kubconfig string)`

Reply:
I was just trying to use the function `clientcmd.BuildConfigFromFlags()`
Then I realized that the input parameter doesn't have to be an output of flag.String, but just a string. Therefore now I modified this to 
`kubeconfig := filepath.Join(os.Getenv("HOME"), ".kube", "config")`

This should be later replaced with `rest.InClusterConfig()` using in-cluster-client-config.